packages/sanity/src/core/components/scroll/scrollContainer.tsx

```diff
@@ -82,7 +81,7 @@ export const ScrollContainer = forwardRef(function ScrollContainer<T extends Ele
 
   return (
     <ScrollContext.Provider value={childContext}>
-      {createElement(as, {'ref': ref, 'data-testid': 'scroll-container', ...rest})}
+      <As data-testid="scroll-container" {...rest} ref={ref} />
```

Comment:
This change allows ScrollContainer to be optimized by React Compiler:
- [before](https://playground.react.dev/#N4Igzg9grgTgxgUxALhASwLYAcIwC4AEcMCAhnggApQBGAyrQQGYwQYEDkAdqVxALRZaYWhwA6XTDnwFgEgkRLkEAUQA2CDAi54ANPIJ4AnlgQF1m7XgAqJhPq4LjpggDFcAd1IwAJgh8ASghMDgpMnt6BwaGGdgQAEtYAsgAylKxYYDFQYAgAwhA6CAAeegY5qkxMCHBljgQVAJLYCDDkaABuCPG8PhrZuUmaEAMIQSESAL7MrOwcSrXikti4hMB0xBBqagVFpdMsbJxgvGjGAPQA+mBoXADmGniFYEsS5wBU7wQAArcUMDw1AR3ucJCVpIQ-q0mKREAQNqxtrs8KRbq10hBMgAeawEEoULg+MDmDRaHS2UwAPgM+O0RIIAHkMGcsYlUhjsdZKbpOKQXgQAD6cQoIrZqDiU2QGPkAfmQJMs5LiQoA1ggjBAmAQAFJ0AAaADpGjoYLcbnALGS8GADCLNts5QQABQILo6eUqN14ACUBAAvJKnb6AwQOhA0D4phIJHBnoQ+Jj-c7g5KoITgmjI1w3p95F9rAALNDErCsDoRhDE0gNPBoNRnIzMNO1NCFZi4Bq5AgeM5Fxx0U7GeF4KA+VvEp6KMgUAhge1qUg0DREQootEwG2OL6NQikNSQAiliDlvxV7ukRvhGArlZcKzEs0VgjVueIhdLsyxnSou-Xyd4No4BVAg7TfEDHDwAsEDQa8mBgsBCCwbwrFnecV2-dc8wIAA5BlrBUeVIOLAhiIqHwCBoRsxyqVoUIQ0dWxrOszjQStDAgQxAOAnAblrZ4QK1BBSXvcDUJIbQsNw-D5R3bs6yBOA9yBXhGzuTptAIAADUCxU0oglJoWFgJ7SDnxw9ougIAARBkknML0eV6Ag+F3AgglhQg6CMHQoNrOA8S9SS8IIghZJ7bZnIgQgMFINVZ1gMxILMITFUhKtaigJTG1fMVF36Qwi2JBD5OchB-AnDiaDMHxCjMEyCyIMAq2JHJMrUPMsN+IoAT3PNQTvYoIXQhD4XnZEf1aJMry8XxxidJhmz4-sxtXCaYBxPFSjpYlLSsCkzD9Tgxw6CUnQMI9MnlUUkVW9cOTAHFuQMabIn8cZ5XcGAZr8KImFZZIUiszpdp0J6uF9OR6i-EbgD5JMOGOjgeR07YeQNdGSAQ6ZDouzcFGhwgSC1Q6KnGf7UiBjoQcIIUuCgbZKSdOntm9aN6iaFo2lrLoekJDRycB4HhJ0QVnPptQeTZQWqeFmmxYZ+aIl8N7omTf1JSJg04FgEgdFZ7MobjQ9kJ0ZF8STCozdKJ1rp2Vd8X1-GjbgIs1B8K3CBJwZhidIN1anZRqHoWgsU9KxGe9HkAG0AF19fKXIVBo2pfZTKV6gUNAtSdFG1AhgwFAUc5zjxZlCByiLXREw52BdusfF1gvC5IEcASIV33ft0oDREGhXzQKqc64W3HcLyYm5b2BHATLADEmaO67dj3keH+c47ZhQKiT6oU79kNIcLghi4IR4ApEmhaA-BpZ4zkjs6Q3W8A9-Pb4USe28Xzu9jwHvhGIAeEBOgflYD2BohBLmLAWUeChx633ftPCAmI57R2AabLueh271w9uvA2m9E7JzwKnf2B9C4EwIAWXoGhbZJhdF6D0Xo04kMPp-UB4D6xgALLQqw0CCCwKbmQoSSZNbaxgI-Dehcs7OgAIRCRfofN+CBW6ODTH4OCd4syHz4bfISBpSA+B8GHHQKRiwElaE6DgFdxQ8goXzBAtseRMMLkhJq6lCIwCgPYJu+NSBYFbggNxHiYgwPjnAxRU81b7y8XiNQBoSAYGPKoL0xiELaDMRY+cSNyGULsfOHhsCYELw7svAgRMcEGHgc6JuWJbagIxCeSaHQ9weL9MAFh6DJjUlfrIYg05VCyydHyBx8xggcHlETHkCNyCkH4BQYqPhRnHHnPwL8a5fyZPRrEyseBJjenyYXLE5wanoINHUisMBOkKH1jskAkwgA)
- [after](https://playground.react.dev/#N4Igzg9grgTgxgUxALhASwLYAcIwC4AEcMCAhnggApQBGAyrQQGYwQYEDkAdqVxALRZaYWhwA6XTDnwFgEggTwBPLAgIBRADYIMCLngAqKhABp5i4wQBiuAO6kYAEwSOASgiZmuCpnYduPLwVlVQIACQMAWQAZSlYsMCCCKDAEAGEIfQQADzwklIR1JiYEODzzAoBJbAQYcjQANwQw3kdtfNTInQgOhHdPCQBfZlZ2DhJSMvFJbFxCYDpiCE1NDKzc4ZY2TjBeNGUAegB9MDQuAHNtPEywaYkDgCoHggABM4oYHk0CB4OJHOkhHetSYkzUi1YKzWeFIZ1qcQgCQAPAYCDkKFxHGANNpdPojKoAHzmdF6LEEADyGH2SIiMQRyIMhJMnFItwIAB9OJkIctNBxCbJzGyAPzIHE6PSGSxcgDWCCUECYBAAUnQABoAOkq+hgZ1OcC0kv0YHMPKWKzFBAAFAgmvpxep7XgAJQEAC8gutbs9BAaEDQjiGEhDXHuT3kzwMAAs0NisKwGoGENjSMk8GhNPslMwoFwymhMsxcMlUgRbPtY946HtlAQ6HgoI5C9jrkQJhQCGALZpSDRtERMjC4TBTd5npVCKRNJACAmIEnnKny6Qc74YIPZlwpdj9cmCGnu5De-21HAh7Dtxu23g6nBZQRzcfH948NGEGgN0xP2BCFgHFKXY9oO+iXrUkYEAAchSBjqOKb5xgQiEFI4BA0DmzbFLUgG-k2hbppm+xoCmigQIod4PjgpwZjcj7KgguI7i+QEkHoEHQbB4qTuWmbfHA07fLwObnI0egEAABk+fLiUQAk0JMD4Vm+B5QfUTQEAAIhSkQaM6LKtAQfBTgQ7iTIQdBKPo74ZnAaLOuxMFwQQ3EVishkQIQGCkPKXawGob5qAxxpAqmZRQAJOZHnyfbtIosbYr+vGGQgLitmRNBqI4mRqEp0ZEGAqbYik4WaJGEFvFknzTpGfzbtkgIgb+9Y9tCYEbu6xYwPYTj9NaTB5gWRa8lCF4jiiaK5GS2JGni0qhB1HDNg0ArWuY84JOKw2rKNV4MmAKLMuY67dc4ARMOKNhdf4Lj9LSUTRBpjQzVKh1cG6cjeI18xsuKACC2ILUtHAslJKwspqEMkL+wwdetY4KOeXBNSQyodQUt10g9T2MfonKGVAKyEtaXAE5oLqhgoVQ1HUGZNC0mLaHdMSPQ0z241yJNg+E90s2zhAc6TRPHddZ0st6HqCijmpwLAJD6OTYafYjTX-nLeDQuiHqlukQ7otaW0a7kCsIzchBwLGmiOIbhBo503TWuLvrEGQFDUPQtDeiyADaAC6CsVKkRQlGUDs+oKH0KAoaDKtaoNk0Kn2RwQBwHGi1KEFFbl2kxWzsObmaOHL5hJyQjafEQFtW7ruSaiINBHmgGWx1wW3G5HgzFwQpewN4ebON+25Bp9gze-nlvWyDLc9n7FPa0HpR4KHEsJ0nKcEFcdlMTQtCnskWCd9HNqq1K1vvZ3Cjd+XY9V+seC18IxCNwg1rH-o1uakI-ZxtGbcKB3iddwQGXXumIPBwiHn-b2r91bVzyBXAu1sZ6K0poHLCIdHbh07srQg0ZWjaC2lrW0zpHTOjDivJO8Dx6wI-rQLMYBoxEKlL-Ag-8sGmzRN8DqUsZYwDVrPKOMcACEDEz4AIvkAnuyRQEDxcJ3VhACGKalII4RwTopTRDjBiWo1oOCZ35CyXBDMEBbRZBHCh-4CqiXgjAKAphz6ySwGXBA1jbFJD-v7ABl9vAYPIUnRRJAMALkKM6DRv49DaN0T2YGBBDFtGMT2Zh-9IEEC9tfCegCmBIPMF4m0nckQG2oQiRctQ-TTlse6YAaTYGDGJGIggSJ-oEEcOQUg-AKCJUcO6MQ4Aez8ERsOK83TZAQ01FDPAwwUYVJRsMA4tSk5IgOAU2+moinJhgHMggCtBguhAIMIAA)